internal/metrics/http.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	clientCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_client_requests_total",
+			Help: "A counter for requests from the wrapped client.",
+		},
+		[]string{"client", "code", "method"},
+	)
+	clientLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_client_request_latency_seconds",
+			Help:    "A histogram of request latencies.",
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"client", "code", "method"},
+	)
+	clientInFlight = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "http_client_in_flight_requests",
```

Comment:
This is an interesting one. So you can see how many requests were in flight for a given time period. 👍